### PR TITLE
Add support to getting statistics for loadbalancer and listener

### PR DESCRIPTION
**Pull request by velp**
In this PR:

1. Add GetStats method for a loadbalancer. See API guide: https://developer.openstack.org/api-ref/load-balancer/v2/#get-load-balancer-statistics. Data structure defined in https://github.com/openstack/octavia/blob/master/octavia/api/v2/types/load_balancer.py#L176
2. Add GetStats method for a listener. See API guide: https://developer.openstack.org/api-ref/load-balancer/v2/#get-listener-statistics. Data structure defined in https://github.com/openstack/octavia/blob/master/octavia/api/v2/types/listener.py#L230

For #1189

### CODE REVIEW

**Comment by jtopjian** [comment]
@velp Thank you for working on this. 

A few initial notes:


> In this PR:

There are a lot of changes happening in this Pull Request and it looks like they can all be implemented independently of each other. Would it be possible to break these up into 3 different PRs?

> Add GetStats method for a loadbalancer

Are you able to reference where in the LBaaS code the data structure is defined? Meaning, can you prove `active_connections` is a valid field? 

> Add GetStats method for a listener

Same here

We have a [Contributor Tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial) that covers these items and might be helpful.

Please let me know if you have any questions. Thank you again.

**Comment by velp** [comment]
@jtopjian I updated this PR, deleted two parts (#1195, #1196).

File: openstack/loadbalancer/v2/listeners/results.go [hunk 1]
```diff
@@ -60,6 +60,23 @@ type Listener struct {
 	ProvisioningStatus string `json:"provisioning_status"`
 }
 
+type StatsTree struct {
```
Comment (jtopjian) [inline openstack/loadbalancer/v2/listeners/results.go]: Let's call this `Stats`

File: openstack/loadbalancer/v2/listeners/results.go [hunk 2]
```diff
@@ -133,3 +150,19 @@ type UpdateResult struct {
 type DeleteResult struct {
 	gophercloud.ErrResult
 }
+
+// GetStatsResult represents the result of a GetStats operation.
+// Call its Extract method to interpret it as a StatsTree.
+type GetStatsResult struct {
```
Comment (jtopjian) [inline openstack/loadbalancer/v2/listeners/results.go]: Let's call this `StatsResult`

File: openstack/loadbalancer/v2/loadbalancers/results.go [hunk 3]
```diff
@@ -58,6 +58,23 @@ type StatusTree struct {
 	Loadbalancer *LoadBalancer `json:"loadbalancer"`
 }
 
+type StatsTree struct {
```
Comment (jtopjian) [inline openstack/loadbalancer/v2/loadbalancers/results.go]: Let's call this `Stats`

File: openstack/loadbalancer/v2/loadbalancers/results.go [hunk 4]
```diff
@@ -124,6 +141,22 @@ func (r GetStatusesResult) Extract() (*StatusTree, error) {
 	return s.Statuses, err
 }
 
+// GetStatsResult represents the result of a GetStats operation.
+// Call its Extract method to interpret it as a StatsTree.
+type GetStatsResult struct {
```
Comment (jtopjian) [inline openstack/loadbalancer/v2/loadbalancers/results.go]: Let's call this `StatsResult`

File: openstack/loadbalancer/v2/listeners/results.go (hunk 1 above)
Comment (velp) [reply openstack/loadbalancer/v2/listeners/results.go]: Fix

File: openstack/loadbalancer/v2/listeners/results.go (hunk 2 above)
Comment (velp) [reply openstack/loadbalancer/v2/listeners/results.go]: Fix

File: openstack/loadbalancer/v2/loadbalancers/results.go (hunk 3 above)
Comment (velp) [reply openstack/loadbalancer/v2/loadbalancers/results.go]: Fix

File: openstack/loadbalancer/v2/loadbalancers/results.go (hunk 4 above)
Comment (velp) [reply openstack/loadbalancer/v2/loadbalancers/results.go]: Fix

**Comment by velp** [comment]
@jtopjian thank you for the review. 
> In addition, could you update the doc.go file in both listeners and loadbalancers to add an example of using this?

Added.

> If you're able to, would you also be able to update the acceptance test at acceptance/openstack/loadbalancer/v2/loadbalancers_test.go with these new functions?

Added.

**Pull request merged**